IanGRap/Squish
# Lobby multiple game modes - Part 1 Menu

Contributors: Alfredo Cuevas

Goal: Create the menu that lets you select the game mode that will be played. Right now this menu does not do anything. If you press start it will simply continue to the lobby like it usually does. This is set up for the future lobby that will adapt to different game modes.

Changes: 
*GameState.cs: Added a new option (GameModeMenu) to LobbyState enum
* LobbyStateManager.cs: Added a new function that changes the LobbyState and moves the camera
* UIOptionPicker.cs: This creates a UI object that lets you scroll through different options. It will eventually have a property that returns the currently selected option. In a future pr, This will be used in the Settings menu, replacing the scrollbar hack.
* Lobby.unity: Added the UI menu between the Title Screen and Lobby. This menu lets you scroll through a list of strings with the names of game modes. 

How To Test: 
* Go into the Lobby. Press play.
* Mess around with the menu. Go back to the Title Screen. Press Start. Does it work as intended?
* Once you press start you should be in the regular lobby character select. From here go back to the Title Screen. 
* How is the transition from the different menus? Is it too quick? 
* How is the Design?
